### PR TITLE
CT: update specification rules to Shanghai in revision ranges

### PR DESCRIPTION
**Pull request by facuMH**
This PR updates a few specification rules so that they can also accept states with Paris and Shanghai revisions.

### CODE REVIEW

File: go/ct/spc/specification.go [hunk 1]
```diff
@@ -1409,7 +1435,7 @@ func getAllRules() []Rule {
 
 	// --- SELFDESTRUCT ---
 
-	selfDestructRevisions := []Revision{R07_Istanbul, R09_Berlin, R10_London}
+	selfDestructRevisions := []Revision{R07_Istanbul, R09_Berlin, R10_London, R11_Paris, R12_Shanghai}
```
Comment (LuisPH3) [inline go/ct/spc/specification.go:1438]: newest supported?

File: go/ct/spc/specification.go [hunk 2]
```diff
@@ -1060,8 +1086,8 @@ func getAllRules() []Rule {
 				RevisionBounds(R07_Istanbul, R09_Berlin),
 				Eq(Status(), st.Running),
 				Eq(Op(Pc()), BASEFEE),
-				Ge(Gas(), 2),
-				Lt(StackSize(), st.MaxStackSize),
+				// Ge(Gas(), 2),
+				// Lt(StackSize(), st.MaxStackSize),
```
Comment (simonlechner) [inline go/ct/spc/specification.go:1090]: Please remove commented out code.

File: go/ct/spc/specification.go [hunk 3]
```diff
@@ -2104,7 +2130,7 @@ func getRulesForAllCallTypes() []Rule {
 
 	res := []Rule{}
 	for _, op := range []OpCode{CALL, CALLCODE, STATICCALL, DELEGATECALL} {
-		for _, rev := range []Revision{R07_Istanbul, R09_Berlin, R10_London} {
+		for _, rev := range []Revision{R07_Istanbul, R09_Berlin, R10_London, R11_Paris, R12_Shanghai} {
```
Comment (LuisPH3) [inline go/ct/spc/specification.go:2133]:
```suggestion
		for i := R07_Istanbul; i <= NewestSupportedRevision; i++  {
```

File: go/ct/spc/specification.go (hunk 2 above)
Comment (LuisPH3) [inline go/ct/spc/specification.go:1090]: dead code

**Review by LuisPH3 — CHANGES_REQUESTED** [review]
Just some small details

**Review by LuisPH3 — DISMISSED** [review]

**Review by simonlechner — DISMISSED** [review]

File: go/ct/spc/specification.go [hunk 4]
```diff
@@ -503,27 +503,34 @@ func getAllRules() []Rule {
...
-		{revision: R10_London, warm: true, config: gen.StorageAssigned, gasCost: 100},
-		{revision: R10_London, warm: true, config: gen.StorageAdded, gasCost: 20000},
-		{revision: R10_London, warm: true, config: gen.StorageAddedDeleted, gasCost: 100, gasRefund: 19900},
-		{revision: R10_London, warm: true, config: gen.StorageDeletedRestored, gasCost: 100, gasRefund: -2000},
-		{revision: R10_London, warm: true, config: gen.StorageDeletedAdded, gasCost: 100, gasRefund: -4800},
-		{revision: R10_London, warm: true, config: gen.StorageDeleted, gasCost: 2900, gasRefund: 4800},
-		{revision: R10_London, warm: true, config: gen.StorageModified, gasCost: 2900},
-		{revision: R10_London, warm: true, config: gen.StorageModifiedDeleted, gasCost: 100, gasRefund: 4800},
-		{revision: R10_London, warm: true, config: gen.StorageModifiedRestored, gasCost: 100, gasRefund: 2800},
+	for rev := R10_London; rev <= R12_Shanghai; rev++ {
```
Comment (HerbertJordan) [inline go/ct/spc/specification.go:508]: Would it be wrong to include `Cancun` already? Maybe even even using `NewestSupportedRevision` as the upper end?

**Review by HerbertJordan — APPROVED** [review]

**Review by LuisPH3 — APPROVED** [review]

**Pull request merged**